TS-5060: Add support for DNAME record type.

jrushford opened the pull request:
This is a patch suggested by @zwoop that adds support for resolving DNAME record types.  I've tested this patch with a delivery service we were trying to deploy that uses DNAME record types.  This patch solves the problem where queries using DNAME failed due to ATS inability to resolve DNAME's.

atsci commented:
FreeBSD build *failed*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1231/ for details.

atsci commented:
Linux build *failed*! See https://ci.trafficserver.apache.org/job/Github-Linux/1125/ for details.

zwoop commented:
Seems two issues here:

1. clang-format
1. FreeBSD doesn't have a T_DNAME ?

jrushford commented:
Taking a look.


Thanks
--
John J. Rushford
IPCDN Engineering
1400 Wewatta Street, Denver Colorado 80202
John_Rushford@cable.comcast.com











From: Leif Hedstrom <notifications@github.com>
Reply-To: apache/trafficserver <reply@reply.github.com>
Date: Thursday, December 1, 2016 at 9:04 AM
To: apache/trafficserver <trafficserver@noreply.github.com>
Cc: John Rushford <John_Rushford@cable.comcast.com>, Author <author@noreply.github.com>
Subject: Re: [apache/trafficserver] TS-5060: Add support for DNAME record type. (#1241)


Seems two issues here:

  1.  clang-format
  2.  FreeBSD doesn't have a T_DNAME ?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/apache/trafficserver/pull/1241#issuecomment-264213697>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AEVybpNvF4mZXXMjhOE6Xu_0J0zXmFVSks5rDvAFgaJpZM4LBjGp>.

PSUdaemon commented:
I'd squash these commits, but this looks good.

jrushford commented:
In FreeBSD T_DNAME is missing from /usr/include/arpa/nameser_compat.h.  However, in /usr/include/arpa/nameser_h, ns_t_dname is defined and In nameser_compat.h, all the T_* dns record types tokens point to the ns_t_* tokens defined in nameser.h  

I changed DNS.cc to use the ns_t_dname token and for consistency, I changed all the other record type tokens to use the ns_t names from nameser.h.  After making these changes, there really is no need for nameser_compat.h in the ATS code so, I removed including it.

atsci commented:
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1126/ for details.

atsci commented:
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1232/ for details.

jrushford commented:
I squashed them, just wanted you to be able to see the commit history.


From: Leif Hedstrom <notifications@github.com>
Reply-To: apache/trafficserver <reply@reply.github.com>
Date: Thursday, December 1, 2016 at 1:26 PM
To: apache/trafficserver <trafficserver@noreply.github.com>
Cc: John Rushford <John_Rushford@cable.comcast.com>, Author <author@noreply.github.com>
Subject: Re: [apache/trafficserver] TS-5060: Add support for DNAME record type. (#1241)


@zwoop commented on this pull request.

Yeh, squash 'em, since otherwise we have potential broken intermediate commits.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/apache/trafficserver/pull/1241#pullrequestreview-11049908>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AEVybtbLuaMzhyAaiB9lHsvGcwxCXqLMks5rDy1_gaJpZM4LBjGp>.

atsci commented:
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1233/ for details.

atsci commented:
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1127/ for details.
